### PR TITLE
feat(dashboards2): Introduce Add Overlay Button

### PR DESCRIPTION
**Pull request by dashed**
Screen Shot 2021-02-05 at 7 21 29 PM

- Rename "Add an overlay" to "Add Y-Axis". This is consistent with how we use the terminology of y-axis in other areas of the product. For example: Discover.
- Use the concept of overlays to refer to group of y-axes. Reintroduced "Add Overlay" button. This button is only shown for line, bars, stacked area, and area charts. 
- Revised "Remove Overlay" button
- Added line separators between query forms
- When removing overlays, if there is only one query left, set its name to be a blank string. This is consistent for when the query name form is not displayed if there is only one query form.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/dashboards/widgetQueryFields.tsx [hunk 1]
```diff
@@ -115,16 +115,16 @@ function WidgetQueryFields({displayType, errors, fields, fieldOptions, onChange}
               borderless
               onClick={event => handleRemove(event, i)}
               icon={<IconDelete />}
-              title={t('Remove this overlay')}
-              label={t('Remove this overlay')}
+              title={t('Remove this Y-Axis')}
+              label={t('Remove this Y-Axis')}
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dashboards/widgetQueryFields.tsx:119]: I think renaming overlay to Y-axis makes sense here. We use y-axis across performance, and discover so I hope that users will have some familiarity with it here too.

File: src/sentry/static/sentry/app/components/dashboards/widgetQueryForm.tsx [hunk 2]
```diff
@@ -80,12 +80,13 @@ class WidgetQueryForm extends React.Component<Props> {
...
-              size="zero"
-              borderless
+              size="small"
+              priority="danger"
               onClick={this.props.onRemove}
               icon={<IconDelete />}
-              title={t('Remove this query')}
-            />
+            >
+              {t('Remove Overlay')}
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dashboards/widgetQueryForm.tsx:88]: The big red button is a bit jarring. Perhaps we can come up with another visual treatment that helps make each overlay/query visually separate so the form doesn't feel so long.

File: src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx [hunk 3]
```diff
@@ -215,10 +215,28 @@ class AddDashboardWidgetModal extends React.Component<Props, State> {
...
+    this.setState(prevState => {
+      const newState = cloneDeep(prevState);
+      newState.queries.push(cloneDeep(newQuery));
+
       return newState;
     });
   };
 
+  canAddOverlay() {
+    return ['line', 'area', 'stacked_area', 'bar'].includes(this.state.displayType);
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx:237]: Should we prune off queries for tables/big number/worldmap when the chart type is changed?

Comment (dashed) [reply src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx:237]: Queries are already pruned in `handleFieldChange()`.

File: src/sentry/static/sentry/app/components/dashboards/widgetQueryForm.tsx (hunk 2 above)
Comment (dashed) [reply src/sentry/static/sentry/app/components/dashboards/widgetQueryForm.tsx:88]: I'll ask @doralchan for guidance here.

File: src/sentry/static/sentry/app/components/dashboards/widgetQueryFields.tsx (hunk 1 above)
Comment (dashed) [reply src/sentry/static/sentry/app/components/dashboards/widgetQueryFields.tsx:119]: I'll be adding some information to the product docs about this.

File: src/sentry/static/sentry/app/components/dashboards/widgetQueryForm.tsx (hunk 2 above)
Comment (dashed) [reply src/sentry/static/sentry/app/components/dashboards/widgetQueryForm.tsx:88]: @doralchan mentioned that she'll follow up with a PR for clean up on Dashboards.

**Review by markstory — APPROVED** [review]
Approving to unblock design revisions.

**Pull request merged**